packages/driver-adapter-utils/src/result.ts

```diff
@@ -40,3 +40,19 @@ export function err<T>(error: Error): Result<T> {
     },
   }
 }
+
+/**
+ * Collect an array of results into an array of values wrapped in a single result.
+ * Equivalent to `results.reduce((a, b) => a.flatMap((a) => b.map((b) => a.concat(b))), ok([] as T[]))`.
+ */
+export function sequence<T>(results: Result<T>[]): Result<T[]> {
```

Comment:
what name would y'all prefer?

Reply:
`flatten`

Reply:
to me `flatten` sounds more like turning multiple nested results into a single one, rather than transposing an array of results into a result of array

Reply:
I don't know any standard name for this kind of transformation, [hoogle](https://hoogle.haskell.org) told me it's called `sequence` in haskell when I typed the signature in, so I used that, but I'm not a fan of the name

Reply:
in rust that would technically be `collect` but it's also very different of how generic it is, but maybe we could call it `collectResults` or `collectIntoResult` here?

Reply:
I know it's technically not exactly flatten but it's a simple good enough approximation to have a quick mental model, considering Result a higher-order wrapping of the value returned. Having said that, I don't have strong opinions. Go with what makes more sense to you.